test/e2e/data/infrastructure-azure/cluster-template-kcp-adoption.yaml

```diff
@@ -0,0 +1,315 @@
+    vnet:
+      name: ${CLUSTER_NAME}-vnet
+  resourceGroup: ${CLUSTER_NAME}
+  subscriptionID: ${AZURE_SUBSCRIPTION_ID}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureMachine
+metadata:
+  name: ${CLUSTER_NAME}-controlplane-0
+  labels:
+    initial: ''  
+spec:
+  dataDisks:
+  - diskSizeGB: 256
+    lun: 0
+    nameSuffix: etcddisk
+  location: ${AZURE_LOCATION}
+  osDisk:
+    diskSizeGB: 128
+    managedDisk:
+      storageAccountType: Premium_LRS
+    osType: Linux
+  sshPublicKey: ${AZURE_SSH_PUBLIC_KEY}
+  vmSize: Standard_D2s_v3
+---      
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: Machine
+metadata:
+  name: ${CLUSTER_NAME}-controlplane-0
+  labels:
+    initial: ''      
+    cluster.x-k8s.io/control-plane: "true"
+spec:
+  version: ${KUBERNETES_VERSION}
+  clusterName: ${CLUSTER_NAME}
+  bootstrap:
+    configRef:
+      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+      kind: KubeadmConfig
+      name: ${CLUSTER_NAME}-controlplane-0
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: AzureMachine
+    name: ${CLUSTER_NAME}-controlplane-0  
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+kind: KubeadmConfig
+metadata:
+  name: ${CLUSTER_NAME}-controlplane-0
+  labels:
+    initial: ''    
+spec:
+  useExperimentalRetryJoin: true
+  initConfiguration:
+    nodeRegistration:
+      kubeletExtraArgs:
+        cloud-config: /etc/kubernetes/azure.json
+        cloud-provider: azure
+      name: '{{ ds.meta_data["local_hostname"] }}'
+  joinConfiguration:
+    nodeRegistration:
+      kubeletExtraArgs:
+        cloud-config: /etc/kubernetes/azure.json
+        cloud-provider: azure
+      name: '{{ ds.meta_data["local_hostname"] }}'
+  clusterConfiguration:
+    apiServer:
+      timeoutForControlPlane: 20m
+      extraArgs:
+        cloud-provider: azure
+        cloud-config: /etc/kubernetes/azure.json
+      extraVolumes:
+        - hostPath: /etc/kubernetes/azure.json
+          mountPath: /etc/kubernetes/azure.json
+          name: cloud-config
+          readOnly: true
+    controllerManager:
+      extraArgs:
+        cloud-provider: azure
+        cloud-config: /etc/kubernetes/azure.json
+        allocate-node-cidrs: "false"
+      extraVolumes:
+        - hostPath: /etc/kubernetes/azure.json
+          mountPath: /etc/kubernetes/azure.json
+          name: cloud-config
+          readOnly: true
+    etcd:
+      local:
+        dataDir: /var/lib/etcddisk/etcd
+  diskSetup:
+    filesystems:
+    - device: /dev/disk/azure/scsi1/lun0
+      extraOpts:
+      - -E
+      - lazy_itable_init=1,lazy_journal_init=1
+      filesystem: ext4
+      label: etcd_disk
+    - device: ephemeral0.1
+      filesystem: ext4
+      label: ephemeral0
+      replaceFS: ntfs
+    partitions:
+    - device: /dev/disk/azure/scsi1/lun0
+      layout: true
+      overwrite: false
+      tableType: gpt
+  mounts:
+  - - LABEL=etcd_disk
+    - /var/lib/etcddisk    
+  files:
```

Comment:
add the secret the old style cause it doesn't generate a secret for an AzureMachine and Machine

Reply:
@CecileRobertMichon is this a bug?

Reply:
hmm it should cc @alexeldeib 

you mean the secret isn't getting generated for the individual `AzureMachine`?  That's what https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/controllers/azurejson_machine_controller.go should do

Reply:
it fails cause it can't get the owner cluster https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/controllers/azurejson_machine_controller.go#L127

Reply:
I think the Owner of the AzureMachine is the Machine and its doesn't check the further in the chain

Reply:
yeah that's definitely a bug. Can you please file an issue?

Reply:
issue here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/874